### DISCUSSION TITLE
No alive nodes found in your cluster

### DISCUSSION THREAD

**Discussion by Chris-Gett** [pimcore/pimcore#10944] [Q&A / Support]
I have in principle a problem when saving objects, it appears the error message "No alive nodes found in your cluster" I have set the Elasticsearch host as here https://github.com/pimcore/pimcore/discussions/9614 in the config.yaml. 

In the parameter pimcore_portal_engine.elasticsearch.host: I have specified "https//myDomain.com:9200" or should there be an IP with port 9200?

when i use "_curl https//myDomain.com:9200_" the connection got refused...

this ist the full error Message:

`Timestamp: Thu Dec 02 2021 10:26:24 GMT+0100 (Mitteleuropäische Normalzeit)
Status: 500 | 
URL: /admin/object/save?task=publish
Method: PUT
Message: No alive nodes found in your cluster
Trace: 
in /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/ConnectionPool/StaticNoPingConnectionPool.php:64
#0 /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php(82): Elasticsearch\ConnectionPool\StaticNoPingConnectionPool->nextConnection()
#1 /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php(99): Elasticsearch\Transport->getConnection()
#2 /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(297): Elasticsearch\Transport->performRequest('POST', '/datahub_restin...', Array, '{"search_after"...', Array)
#3 /home/xxx/xxx/vendor/react/promise/src/FulfilledPromise.php(28): Elasticsearch\Connections\Connection->Elasticsearch\Connections\{closure}(Array)
#4 /home/xxx/xxx/vendor/ezimuel/ringphp/src/Future/CompletedFutureValue.php(55): React\Promise\FulfilledPromise->then(Object(Closure), NULL, NULL)
#5 /home/xxx/xxx/vendor/ezimuel/ringphp/src/Core.php(341): GuzzleHttp\Ring\Future\CompletedFutureValue->then(Object(Closure), NULL, NULL)
#6 /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(345): GuzzleHttp\Ring\Core::proxy(Object(GuzzleHttp\Ring\Future\CompletedFutureArray), Object(Closure))
#7 /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(241): Elasticsearch\Connections\Connection->Elasticsearch\Connections\{closure}(Array, Object(Elasticsearch\Connections\Connection), Object(Elasticsearch\Transport), Array)
#8 /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php(110): Elasticsearch\Connections\Connection->performRequest('POST', '/datahub_restin...', Array, '{"search_after"...', Array, Object(Elasticsearch\Transport))
#9 /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Client.php(1946): Elasticsearch\Transport->performRequest('POST', '/datahub_restin...', Array, Array, Array)
#10 /home/xxx/xxx/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Client.php(1365): Elasticsearch\Client->performRequest(Object(Elasticsearch\Endpoints\Search))
#11 /home/xxx/xxx/vendor/pimcore/data-hub-simple-rest/src/Service/IndexHandler.php(408): Elasticsearch\Client->search(Array)
#12 /home/xxx/xxx/vendor/pimcore/data-hub-simple-rest/src/Service/IndexHandler.php(385): Pimcore\Bundle\DataHubSimpleRestBundle\Service\IndexHandler->doLoadAllIds('datahub_restind...', Array)
#13 /home/xxx/xxx/vendor/pimcore/data-hub-simple-rest/src/Service/IndexService.php(318): Pimcore\Bundle\DataHubSimpleRestBundle\Service\IndexHandler->getAllIdsFromIndexByOriginalPath('datahub_restind...', '/Product Data/C...')
#14 /home/xxx/xxx/vendor/pimcore/data-hub-simple-rest/src/EventSubscriber/ChangedEventSubscriber.php(59): Pimcore\Bundle\DataHubSimpleRestBundle\Service\IndexService->invalidateIndexByOriginalPath('/Product Data/C...', 'Category')
#15 /home/xxx/xxx/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(117): Pimcore\Bundle\DataHubSimpleRestBundle\EventSubscriber\ChangedEventSubscriber->updateElement(Object(Pimcore\Event\Model\DataObjectEvent), 'pimcore.dataobj...', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher))
#16 /home/xxx/xxx/vendor/symfony/event-dispatcher/EventDispatcher.php(230): Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(Object(Pimcore\Event\Model\DataObjectEvent), 'pimcore.dataobj...', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher))
#17 /home/xxx/xxx/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\Component\EventDispatcher\EventDispatcher->callListeners(Array, 'pimcore.dataobj...', Object(Pimcore\Event\Model\DataObjectEvent))
#18 /home/xxx/xxx/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(151): Symfony\Component\EventDispatcher\EventDispatcher->dispatch(Object(Pimcore\Event\Model\DataObjectEvent), 'pimcore.dataobj...')
#19 /home/xxx/xxx/vendor/pimcore/pimcore/models/DataObject/AbstractObject.php(767): Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(Object(Pimcore\Event\Model\DataObjectEvent), 'pimcore.dataobj...')
#20 /home/xxx/xxx/vendor/pimcore/pimcore/models/DataObject/Concrete.php(739): Pimcore\Model\DataObject\AbstractObject->save(Array)
#21 /home/xxx/xxx/vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php(1364): Pimcore\Model\DataObject\Concrete->save()
#22 /home/xxx/xxx/vendor/symfony/http-kernel/HttpKernel.php(156): Pimcore\Bundle\AdminBundle\Controller\Admin\DataObject\DataObjectController->saveAction(Object(Symfony\Component\HttpFoundation\Request))
#23 /home/xxx/xxx/vendor/symfony/http-kernel/HttpKernel.php(78): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)
#24 /home/xxx/xxx/vendor/symfony/http-kernel/Kernel.php(199): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#25 /home/xxx/xxx/public/index.php(36): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request))
#26 {main}`

If I disable the Data Hub Simple Rest Bundle I can store objects or other data types without error message. 

Also I can't install the Pimcore Portal Engine Bundle, because during the installation the same message (as above) appears. So there seems to be an error somewhere in parameters of Elasticsearch.

Is there anything else to configure besides the parameters? ist there a best practise?

I´ve also read the Docs about it here:
https://pimcore.com/docs/data-hub-simple-rest/current/Installation.html
https://pimcore.com/docs/portal-engine/current/Installation/Installation.html#page_Required-Parameter

 I am getting stuck right now...

**Comment by fashxp** [comment]
first step is to check if the elasticsearch cluster is reachable from your php server ...

**Reply by Chris-Gett** [reply]
The way I read the documentation, PIMCORE does not require a separate instance of an Elasticsearch server.  The basic settings you need to set in Pimcore are the host and the prefix. So the host should be accessible via https://myDomain.com:port right? I have set both as above. To test the cluster I use the curl command as described above.  Do I have to read over further documentations to use these basic functions at all?

**Reply by fashxp** [reply]
but you write already, that the connection got refused with your curl command ... thus most likely also pimcore cannot connect to the ES cluster.

**Reply by Chris-Gett** [reply]
that is exactly my problem, I have just no approach where I can look further, if the settings are correct. I can´t find any Hints for debugging this in the Documentation.

**Reply by fashxp** [reply]
this has nothing to do with Pimcore ... this is the basic connection to elastic search... 
pimcore does not come with an integrated elastic search instance. you need to provide it, as you need to provide the database and the php ... maybe that is your missing link?

**Reply by Chris-Gett** [reply]
okay, that´s really good to knew. this is indeed the missing piece of the puzzle. if possible, a note in the documentation would be quite helpful here. I personally assumed that everything necessary is installed with the appropriate bundles and only the above-mentioned configuration is necessary.

**Reply by fashxp** [reply]
Could you please provide a PR with the comment where you would have expected it?
Thx very much!
